draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1991,36 +2012,199 @@ conditions.
 
 # Suites for Hashing {#suites}
 
-The following table lists recommended algorithms for different curves and
-encodings. To select a suitable algorithm, choose the encoding associated with
-the target curve. For example, Elligator 2 is the recommended encoding for
-Curve25519, whereas simplified SWU is the recommended encoding for P-256.
-When the hash function is to be used in a protocol whose security is proven in the
-random oracle model, applications SHOULD use the Random Oracle construction
-given in {{roadmap}}.
-
-A suite is a bundle of algorithms that provides concrete recommendations for
-hashing bit strings into points of specific elliptic curve groups. Each suite is
-a tuple (E, H, f, ROM) such that
-
--   E, is the elliptic curve group.
--   H, is the cryptographic hash function used by hash\_to\_base.
--   f, is an encoding function compatible with E.
--   ROM, is a boolean flag indicating whether or not to use the random oracle construction.
-
-This document describes the following set of ciphersuites
-
-| Suite ID | E | H | f | ROM |
-|----------|---|---|---|-----|
-| H2C-0001 | P256         | SHA256 | Simplified SWU           | True |
-| H2C-0002 | P384         | SHA512 | Icart                    | True |
-| H2C-0003 | curve25519   | SHA512 | Elligator 2              | True |
-| H2C-0004 | curve448     | SHA512 | Elligator 2              | True |
-| H2C-0005 | edwards25519 | SHA512 | Elligator 2              | True |
-| H2C-0006 | edwards448   | SHA512 | Elligator 2              | True |
-| H2C-0007 | SECP256K1    | SHA512 | Shallue-van de Woestijne | True |
-| H2C-0008 | BLS12381     | SHA512 | Simplified SWU           | True |
+This section lists recommended suites for hashing to standard elliptic curves.
+
+A suite fully specifies the procedure for hashing bit strings to
+points on a specific elliptic curve group.
+Each suite comprises the following parameters:
+
+- Suite ID, a short name used to refer to a given suite.
+- E, the target elliptic curve over a field F.
+- p, the characteristic of the field F.
+- m, the extension degree of the field F.
+- H, the hash function used by hash\_to\_base.
+- W, the number of evaluations of H in hash\_to\_base.
+- f, a mapping function given in {{mappings}}.
+- h\_eff, the value by which the output of f is multiplied
+  for the purpose of clearing the cofactor ({{cofactor-clearing}}).
+
+In addition to the above parameters, the mapping f may require
+additional parameters Z, rational\_map, M, E', and/or iso\_map.
+These are specified when applicable.
+
+Suites whose ID ends in "-RO" use the hash\_to\_curve procedure of {{roadmap}}.
+All other suites use the encode\_to\_curve procedure from that section.
+Applications whose security requires a random oracle MUST use a "-RO" suite.
+
+The below table lists the curves for which suites are defined and
+the subsection that gives the corresponding parameters.
```

Comment:
Should we leave a note that newly standardized curves SHOULD provide similar suites?